pallets/aleph/src/mock.rs

```diff
@@ -173,18 +174,26 @@ pub fn new_test_ext(authorities: &[(u64, u64)]) -> sp_io::TestExternalities {
     t.into()
 }
 
-pub(crate) fn run_session(n: u64) {
-    while System::block_number() < n {
+pub(crate) fn run_session(n: u32) {
+    for i in Session::current_index()..n {
         Session::on_finalize(System::block_number());
         Aleph::on_finalize(System::block_number());
         System::on_finalize(System::block_number());
 
+        let parent_hash = if System::block_number() > 1 {
+            System::finalize().hash()
```

Comment:
what are the reasons behind introducing `System::finalize().hash()` and `set_block_number` (below)?

Reply:
We don't need to run all blocks from scratch on every call of this method.